docs/DeveloperGuide.md

```diff
@@ -2,36 +2,524 @@
+the command keyword to a predefined set of commands.
+
+- **Arguments Extraction:** For commands that require additional information, the Parser should correctly extract and format
+arguments, ensuring they are ready for command execution.
+
+- **Error Handling:** In cases where the input does not match any recognized command or has formatting errors, the Parser
+should generate appropriate error messages.
+
+
+#### Implementation and rationale
+
+The Parser class is designed to handle these considerations through a well-structured parsing process. Here's how it
+works:
+
+1. Splitting User Input: The Parser takes the full user input and splits it into two parts: the command word and the
+remaining arguments.
+
+2. Command Recognition: It matches the command word to a predefined set of commands. If a valid command is recognized, it
+proceeds to the next step.
+
+3. Arguments Extraction: Depending on the specific command, the Parser may further parse and extract arguments. For
+instance, for the "AddCommand," it extracts drug-related details like name, expiry date, serial number, and quantity.
+
+4. Command Creation: The Parser creates an instance of the appropriate Command class, passing along any required arguments.
+This encapsulates the user's request in an executable command object.
+
+5. Error Handling: If the user input does not match any recognized command or has formatting errors, the Parser generates
+an "IncorrectCommand" with an error message, providing feedback to the user.
+
+By structuring the parsing process this way, the application ensures that user input is correctly interpreted and
+translated into executable commands for the subsequent phases of the application.
+
+The parser class uses the below method to achieve its functionality
+
+- `parseCommand(String userInput)` - This method takes the user's input as a parameter and processes it to identify the
+  command keyword and any associated arguments. It then recognizes the specific command and prepares it for execution.
+
+Given below is an example of how the login system class works.
+
+Suppose a user enters the command `add /n Paracetamol /d 2023-12-31 /s ABC123 /q 100`. The Parser first splits this input
+into the command word "add" and the arguments. It then recognizes the "add" command, extracts the drug details (name,
+expiry date, serial number, and quantity), and creates an instance of the "`AddCommand`" with these details. If the user
+enters an invalid command or incorrect formatting, the Parser provides feedback to guide the user, ensuring a seamless
+interaction between the user and the application.
+
+The Parser is a crucial component that forms the bridge between user intentions and the core functionality of the stock
+management system.
+
+The following sequence diagram shows how the parser class works when the program is running.
+<img src="UML Diagrams/ParserDiagram.png" width="500">
+
```

Comment:
I think the creating of incorrect command is wrong. There should be a constructor and should return from the IncorrectCommand class instead of a self invocation.